File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/job/meta/StopJobResponse.scala
package com.webank.eggroll.core.deepspeed.job.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class StopJobResponse(sessionId: String)

object StopJobResponse {
  implicit def serialize(src: StopJobResponse): Array[Byte] = {
    Deepspeed.StopJobResponse.newBuilder()
      .setSessionId(src.sessionId)
      .build()
      .toByteArray
  }

  implicit def deserialize(byteString: ByteString): StopJobResponse = {
    val proto = Deepspeed.StopJobResponse.parseFrom(byteString)
    StopJobResponse(sessionId = proto.getSessionId)
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/StopJobResponse.java
package com.eggroll.core.pojo;

import com.webank.eggroll.core.meta.Deepspeed;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class StopJobResponse implements RpcMessage{

    Logger log = LoggerFactory.getLogger(StopJobResponse.class);
    private String sessionId;

    @Override
    public byte[] serialize() {
        Deepspeed.StopJobResponse.Builder builder = Deepspeed.StopJobResponse.newBuilder();
        builder.setSessionId(this.sessionId);
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {
        try {
            Deepspeed.StopJobResponse protoEntity = Deepspeed.StopJobResponse.parseFrom(data);
            this.sessionId = protoEntity.getSessionId();
        } catch (Exception e) {
            log.error("deserialize error : ", e);
        }
    }
}
